File: 1365. How Many Numbers Are Smaller Than the Current Number/_1365.py
class Solution:
    def smallerNumbersThanCurrent(self, nums):
        myDict = dict()
        nums2 = sorted(nums)
        for i,v in enumerate(nums2):
            tmp = myDict.get(v, -1)
            if tmp == -1:
                myDict[v] = i

        res = list()
        for i in nums:
            res.append(myDict[i])

        return res

a = Solution()
print(a.smallerNumbersThanCurrent([7,7,7,7]))

File: 1365. How Many Numbers Are Smaller Than the Current Number/1365.go
package leetcode

import(
	"sort"
)

func smallerNumbersThanCurrent(nums []int) []int {
	nums2 := make([]int, len(nums), len(nums))
	copy(nums2,nums)

	sort.Ints(nums2)

	myDict := make(map[int]int, 0)

	for i,v := range nums2{
		if _, ok := myDict[v]; !ok{
			myDict[v] = i
		}
	}
	
	res := make([]int, 0, len(nums))

	for _,v := range nums{
		value,_ := myDict[v]
		res = append(res, value)
	}

	return res
}